internal/docstore/drivertest/drivertest.go

```diff
@@ -811,24 +857,24 @@ func testQuery(t *testing.T, coll *ds.Collection) {
 	}
 
 	// For limit, we can't be sure which documents will be returned, only their count.
-	limitQ := coll.Query().Where("n", ">=", -4).Limit(2)
-	got := mustCollect(ctx, t, limitQ.Get(ctx, KeyField))
+	limitQ := coll.Query().Limit(2)
+	got := mustCollectHighScores(ctx, t, limitQ.Get(ctx))
 	if len(got) != 2 {
 		t.Errorf("got %v, wanted two documents", got)
 	}
 }
 
-func forEach(ctx context.Context, iter *ds.DocumentIterator, f func(docmap) error) error {
+func forEach(ctx context.Context, iter *ds.DocumentIterator, create func() interface{}, handle func(interface{}) error) error {
```

Comment:
Can `create func() interface{}` be a `doc interface{}`?

Reply:
Yes, but it would require more reflection than I thought was worth it, since this isn't user-facing API.

Reply:
(We can't just use the doc, we have to make a new one for each value.)